[16.0][MIG] pos_session_pay_invoice: Migration to version 16.0

carlos-lopez-tecnativa opened the pull request:
@tecnativa TT49807

etobella committed:
[ADD] PoS session pay invoice module

JordiBForgeFlow committed:
Migration of pos_session_pay_invoice to 11.0

bishalgit committed:
Fix account-payment's github repo url

Commit:
[12.0][MIG] pos_session_pay_invoice

danimv5 committed:
Added translation using Weblate (Spanish)

jguenat committed:
Added translation using Weblate (French)

acsonefho committed:
[IMP] pos_session_pay_invoice: black, isort, prettier

acsonefho committed:
[13.0][MIG] pos_session_pay_invoice

olgamarcocb committed:
[MIG] pos_session_pay_invoice: Ending migration to 13.0

olgamarcocb committed:
[IMP] pos_session_pay_invoice: black, isort, prettier

olgamarcocb committed:
[MIG] pos_session_pay_invoice: Migration to 14.0

oca-travis committed:
[UPD] Update pos_session_pay_invoice.pot

mymage committed:
Added translation using Weblate (Italian)

Commit:
[14.0][IMP] pos_session_pay_invoice: Set `target=new` on views

Commit:
[FIX] `search_paid_order_ids` must exclude orders createds with this module

Commit:
[UPD] Update pos_session_pay_invoice.pot

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (30 of 30 strings)

Translation: pos-14.0/pos-14.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_session_pay_invoice/it/

OCA-git-bot committed:
pos_session_pay_invoice 14.0.1.1.0

anmarmo1 committed:
Translated using Weblate (Spanish)

Currently translated at 86.6% (26 of 30 strings)

Translation: pos-14.0/pos-14.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_session_pay_invoice/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (30 of 30 strings)

Translation: pos-14.0/pos-14.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_session_pay_invoice/es/

carlos-lopez-tecnativa committed:
[IMP] pos_session_pay_invoice: pre-commit stuff

carlos-lopez-tecnativa committed:
[MIG] pos_session_pay_invoice: Migration to 16.0

legalsylvain commented on pos_session_pay_invoice/readme/ROADMAP.rst:
```diff
@@ -0,0 +1,5 @@
+* Cannot pay invoices in a different currency than that defined in the journal
+  associated to the payment method used to pay/collect payment.
+
+* Should depend on `pos_invoicing` but it requires a refactoring of `pos_invoicing`.
+  It will be improved when migrating to 13.0
```
This part is obsolete, as the invoices are now marked as paid.

carlos-lopez-tecnativa replied:
removed

legalsylvain reviewed (commented):
Hi @carlos-lopez-tecnativa. 


Thanks for porting this module tecnically, I don't understand why 
- some button open a wizard that inherit from cash.pay.invoice,
- and in other button, it open a new model pos.box.cash.invoice.in / pos.box.cash.invoice.out.

could you elaborate ?

carlos-lopez-tecnativa commented:
> Hi @carlos-lopez-tecnativa.
> 
> Thanks for porting this module tecnically, I don't understand why
> 
> * some button open a wizard that inherit from cash.pay.invoice,
> * and in other button, it open a new model pos.box.cash.invoice.in / pos.box.cash.invoice.out.
> 
> could you elaborate ?

Hi @legalsylvain,

In this module, I have only migrated and made minimal changes for now. However, if you consider it necessary, I can refactor it similarly to [this PR](https://github.com/OCA/account-payment/pull/743) to inherit from the new wizard `cash.pay.invoice` for customer/vendor invoices instead of using two separate wizards.

Before the referenced PR, these wizards (`cash.invoice.in`, `cash.invoice.out`) inherited from `cash.box.out`, but as it no longer exists, I created a single, more generic wizard called `cash.pay.invoice`. In this module, the wizard that depends on `cash.invoice.in` is updated to inherit from `cash.pay.invoice`. However, the wizards `pos.box.cash.invoice.out` and `pos.box.cash.invoice.in` do not inherit from any, so I didn't touch them.

Let me know if the refactor is welcome.

legalsylvain commented:
Hi @carlos-lopez-tecnativa. 
Sorry. I didn't understood. I tested a little locally.
Functionaly, why there are two different ways to do the same thing ?

carlos-lopez-tecnativa commented:
> Hi @carlos-lopez-tecnativa. Sorry. I didn't understood. I tested a little locally. Functionaly, why there are two different ways to do the same thing ?

@legalsylvain  That was the behavior in the previous version; there were several wizards to do each thing. Now, based on my previous comment and the referenced PR(https://github.com/OCA/account-payment/pull/743), we could simplify this. Let me know if that works for you. @pedrobaeza what do you think?

pedrobaeza commented:
Yes, please simplify it.

legalsylvain commented:
Agree with Pedro. Let me know when you want review.
Thanks for the work !

carlos-lopez-tecnativa commented:
@pedrobaeza @legalsylvain I have updated the code to inherit functionality from the `cash.pay.invoice` wizard and removed the older wizards. Please review the changes and let me know if they work fine for you.

pedrobaeza commented:
/ocabot migration pos_session_pay_invoice

pedrobaeza commented on pos_session_pay_invoice/README.rst:
```diff
@@ -0,0 +1,104 @@
+Configuration
+=============
+
+#.  Go to *Point of Sale / Configuration / Point of Sale* and activate the
```
I think this text is outdated. There's no such option.

carlos-lopez-tecnativa replied:
I updated the README. Please let me know if it is OK now.

pedrobaeza reviewed (commented):
I can't find the buttons specified in the README in the POS session.

pedrobaeza commented:
I don't understand why limiting the invoice payment to cash methods. Someone can come to your physical store to pay an invoice, and choose to pay with credit card. I think that limitation should be gone.

pedrobaeza commented:
And anyway, even adding a cash payment method, I don't find the buttons referred in the README

carlos-lopez-tecnativa commented:
> And anyway, even adding a cash payment method, I don't find the buttons referred in the README

@pedrobaeza the buttons are here. Could you attach an image, please?
![image](https://github.com/user-attachments/assets/fb503020-2723-4775-862d-e57914bd7d0e)

![image](https://github.com/user-attachments/assets/5e218dd7-b256-4c01-a37f-3ed355869a17)

pedrobaeza commented:
I was looking in the PoS session "frontend". In any moment the README mentions that you should go to the session backend form.

carlos-lopez-tecnativa commented:
> I don't understand why limiting the invoice payment to cash methods. Someone can come to your physical store to pay an invoice, and choose to pay with credit card. I think that limitation should be gone.

This behavior is from the previous version. Do you want us to change it to allow the user to select any payment method available on POS? (But only if the payment method has a journal set.) Please confirm so I can adapt the code.

pedrobaeza commented:
OK, I see the functionality very limited having to go to backend, so let's not invest more time in this. Please clarify the usage README though.

carlos-lopez-tecnativa commented:
> OK, I see the functionality very limited having to go to backend, so let's not invest more time in this. Please clarify the usage README though.

Odoo, in the new version, has moved most functionalities to the frontend. This implies more effort to change it to the frontend. The task for now is to migrate the current module.
I have updated the readme. Please let me know if it is clearer now. Thanks.

pedrobaeza reviewed (approved):
Code and functional review

carlos-lopez-tecnativa commented:
@legalsylvain, could you please review again?

legalsylvain commented:
Hi @carlos-lopez-tecnativa. Thanks for the ping ! I just come back from hollidays today. I take a look this week. 

thanks !

carlos-lopez-tecnativa commented:
> Hi @carlos-lopez-tecnativa. Thanks for the ping ! I just come back from hollidays today. I take a look this week.
> 
> thanks !

@legalsylvain I hope you had a great holiday! Would you kindly take another look when you have a chance? I'd appreciate any updates or feedback you may have.

ivantodorovich reviewed (approved):
code review LG

pedrobaeza commented:
/ocabot merge nobump

pedrobaeza commented:
/ocabot merge nobump

legalsylvain commented on pos_session_pay_invoice/models/pos_order.py:
```diff
@@ -0,0 +1,25 @@
+    _inherit = "pos.order"
+
+    @api.model
+    def search_paid_order_ids(self, config_id, domain, limit, offset):
```
Hi. Why this change in that module. I'm not sure to understand why this overload is interesting, but my question is why in this module ?

carlos-lopez-tecnativa replied:
This code is present in V14. Please refer to this PR for the changes that were added: https://github.com/OCA/pos/pull/1006.
Let me know if this code is useful or if it should be removed.

carlos-lopez-tecnativa replied:
Reading the PR, the error is raised with orders created in this module, but now orders aren’t created; only `account.bank.statement.line` are created. So, I think the code is unused. Let me know what you think, please.

legalsylvain replied:
indeed, I think this part of code is now obsolete.

carlos-lopez-tecnativa replied:
Done, I removed the code.

legalsylvain replied:
thanks !

legalsylvain commented on pos_session_pay_invoice/readme/USAGE.rst:
```diff
@@ -0,0 +1,8 @@
+#.  Go to *Point of Sale / Dashboard* and create and open or access to an
+    already open POS Session.
+#.  Open the POS Session form view on the Backend.
+#.  Press the button **Pay Invoice** to pay a Supplier Invoice or a Customer
+    Refund. It will be paid using Cash.
```
```suggestion
#.  Press the button **Pay Supplier Invoice** to pay a Supplier Invoice or **Pay Refund** for a Customer
    Refund. It will be paid using Cash.
```

legalsylvain commented on pos_session_pay_invoice/readme/USAGE.rst:
```diff
@@ -0,0 +1,8 @@
+#.  Open the POS Session form view on the Backend.
+#.  Press the button **Pay Invoice** to pay a Supplier Invoice or a Customer
+    Refund. It will be paid using Cash.
+#.  Select **Collect Payment from Invoice** in to receive a payment of an
```
```suggestion
#.  Select **Get Payment from Invoice** to receive a payment of an
```

legalsylvain reviewed (commented):
Code review : 
`` 21 files changed, 219 insertions(+), 471 deletions(-)
``
- Nice simplification ! 
- a question regarding the pos_order.py file. 
- some minor adaptation in the readme files, to match with the UI.

Functional review : LGTM.

Thanks a lot for your work, and sorry for the delay.

legalsylvain commented:
Thanks !

legalsylvain commented:
/ocabot merge nobump

carlos-lopez-tecnativa commented:
I have no idea why the test failed. It works fine on my local environment. Does anyone have any ideas? The test that failed is in other modules.

![image](https://github.com/user-attachments/assets/440a4b8d-e43c-48f3-ab80-ea80379b0de4)


![image](https://github.com/user-attachments/assets/5657072d-65e1-46eb-8542-444362d94bbb)

carlos-lopez-tecnativa commented:
I can see the error is general in the tour; other PRs have the same error in some OCA repos


https://github.com/OCA/calendar/actions/runs/10941425935/job/30376034658#step:8:159

https://github.com/OCA/pos/actions/runs/10921083459/job/30312371146#step:8:1086

https://github.com/OCA/e-commerce/actions/runs/10902558739/job/30254730888#step:8:306

carlos-lopez-tecnativa commented:
@legalsylvain @pedrobaeza the PR is ready. The error in the tests is resolved in https://github.com/OCA/oca-ci/pull/78.

pedrobaeza commented:
/ocabot merge nobump
